scripts/generate.go

```diff
@@ -0,0 +1,374 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"html/template"
+	"os"
+	"os/signal"
+	"path"
+	"path/filepath"
+	"sort"
+	"strings"
+	"syscall"
+)
+
+const (
+	readmeTitle              = "Google GitHub Actions - Example Workflows"
+	propertiesDirName string = "properties"
+)
+
+var (
+	pathPtr    = flag.Bool("path", false, "starter workflow path")
+	starterPtr = flag.Bool("starter", false, "starter workflow")
+	typePtr    = flag.String("type", "deployments", "starter workflow type")
+
+	propertiesTemplPath string = path.Join("templates", "workflow.properties.tmpl.json")
+	rootWorkflowPath    string = path.Join("workflows")
+	workflowConfigPath  string = path.Join("workflow.config.json")
+	readmeTmplatePath   string = path.Join("templates", "README.tmpl.md")
+	readmeOutputPath    string = path.Join(defaultEnv("OUTPUT_PATH", "README.md"))
+)
+
+func main() {
+	ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+	defer cancel()
+
+	flag.Parse()
+
+	if err := realMain(ctx); err != nil {
+		cancel()
+		fmt.Fprintf(os.Stderr, "%s\n", err)
+		os.Exit(1)
+	}
+}
+
+func realMain(ctx context.Context) error {
+	args := flag.Args()
+	if len(args) <= 0 {
+		return fmt.Errorf("expected arguments, got 0")
+	}
+
+	command := args[0]
+
+	if strings.EqualFold(command, "workflow") {
+		return generateWorkflow(ctx, args)
+	}
+
+	if strings.EqualFold(command, "readme") {
+		return generateReadme(ctx)
+	}
+
+	return fmt.Errorf("invalid command: %s", command)
+}
+
+// generateWorkflow handles the creation of new workflow files
+func generateWorkflow(ctx context.Context, args []string) error {
+	if len(args) != 2 {
+		return fmt.Errorf("expected 2 arguments, got %d: %q", len(args), args)
+	}
```

Comment:
Are we handling `--starter`, `--type` in a follow up?

Reply:
They are defaulted it not provided, see var definitions at top

starter defaults to false
type defaults to deployments